src/dsa/rpo_falcon512/signature.rs

```diff
@@ -47,72 +47,71 @@ pub struct Signature {
     pub(super) pk: PublicKeyBytes,
     pub(super) sig: SignatureBytes,
 
-    // Cached polynomial decoding for public key and signatures
-    pub(super) pk_polynomial: OnceCell<Polynomial>,
-    pub(super) sig_polynomial: OnceCell<Polynomial>,
+    // Cached polynomial decoding for public key and signature
+    pub(super) pk_poly: Polynomial<FalconFelt>,
+    pub(super) sig_poly: Polynomial<FalconFelt>,
```

Comment:
Yes, it makes sense. The reason for keeping the bytes representation was because during the signing procedure, we do have to encode the signature into bytes anyway and thus we save some amount of work by storing the resulting byte encoding.
I have changed it now as the clarity benefits outweigh the above logic.